staker/staker.go

```diff
@@ -425,6 +575,13 @@ func (s *Staker) Start(ctxIn context.Context) {
 		if err != nil {
 			log.Warn("error updating latest wasm module root", "err", err)
 		}
+		switchedToBoldProtocol, err := s.checkAndSwitchToBoldStaker(ctxIn)
+		if err != nil {
+			log.Error("staker: error in checking switch to bold staker", "err", err)
+		}
+		if switchedToBoldProtocol {
+			s.StopAndWait()
```

Comment:
return

Reply:
Also, I think this will deadlock, even with the return.
use StopOnly to stop without waiting